packages/node/src/concordium/api.concordium.ts

```diff
@@ -0,0 +1,323 @@
+        events.push(tx.transfer);
+        events.push(tx.memo);
+        break;
+      }
+      case TransactionKindString.TransferWithSchedule: {
+        events.push(tx.event);
+        break;
+      }
+      case TransactionKindString.TransferWithScheduleAndMemo: {
+        events.push(tx.transfer);
+        events.push(tx.memo);
+        break;
+      }
+      case TransactionKindString.EncryptedAmountTransfer: {
+        events.push(tx.removed);
+        events.push(tx.added);
+        break;
+      }
+      case TransactionKindString.EncryptedAmountTransferWithMemo: {
+        events.push(tx.removed);
+        events.push(tx.added);
+        events.push(tx.memo);
+        break;
+      }
+      case TransactionKindString.DeployModule: {
+        events.push(tx.moduleDeployed);
+        break;
+      }
+      case TransactionKindString.InitContract: {
+        events.push(tx.contractInitialized);
+        break;
+      }
+      case TransactionKindString.Update: {
+        events.push(...tx.events);
+        break;
+      }
+      case TransactionKindString.RegisterData: {
+        events.push(tx.dataRegistered);
+        break;
+      }
+      case TransactionKindString.TransferToPublic: {
+        events.push(tx.removed);
+        events.push(tx.added);
+        break;
+      }
+      case TransactionKindString.TransferToEncrypted: {
+        events.push(tx.added);
+        break;
+      }
+      case TransactionKindString.AddBaker: {
+        events.push(tx.bakerAdded);
+        break;
+      }
+      case TransactionKindString.RemoveBaker: {
+        events.push(tx.bakerRemoved);
+        break;
+      }
+      case TransactionKindString.UpdateBakerKeys: {
+        events.push(tx.bakerKeysUpdated);
+        break;
+      }
+      case TransactionKindString.UpdateBakerStake: {
+        events.push(tx.bakerStakeChanged);
+        break;
+      }
+      case TransactionKindString.UpdateBakerRestakeEarnings: {
+        events.push(tx.bakerRestakeEarningsUpdated);
+        break;
+      }
+      case TransactionKindString.ConfigureBaker: {
+        events.push(...tx.events);
+        break;
+      }
+      case TransactionKindString.ConfigureDelegation: {
+        events.push(...tx.events);
+        break;
+      }
+      case TransactionKindString.UpdateCredentialKeys: {
+        events.push(tx.keysUpdated);
+        break;
+      }
+      case TransactionKindString.UpdateCredentials: {
+        events.push(tx.credentialsUpdated);
+        break;
+      }
+      default:
+    }
+
+    return events.map((evt, i) => {
+      return {
+        ...evt,
+        id: i,
+        block: null,
+        transaction: tx,
+      } as ConcordiumTransactionEvent;
+    });
+  }
+  /* eslint-enable complexity */
+
+  async getAndWrapTransactions(
+    blockHash: string,
+  ): Promise<ConcordiumTransaction[]> {
+    const transactions = await streamToList(
+      this.client.getBlockTransactionEvents(blockHash),
```

Comment:
Is this function just poorly named? It reads like it gets the events not the transactions.

Reply:
yeah it is just poorly named, it returns a stream of block item summaries (transactions)